### PR TITLE
feat: publish command

### PR DESCRIPTION
**Pull request by EvolutionX-10**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -38,4 +39,10 @@ program
 	.description('Easy way to add extra things in your sern project')
 	.action(extra);
 
+program
+	.command(publish.name)
+	.description('New way to manage your slash commands')
+	.argument('<name>', 'name of the file without extension')
```
Comment (jacoobes) [inline src/index.ts:45]: we would want to make <name> argument a glob. its a bit repetitive to publish single commands

File: src/utilities/getConfig.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { findUp } from 'find-up';
+
+export async function getConfig(): Promise<sernConfig> {
+	const sernLocation = await findUp('sern.config.json');
+
+	if (!sernLocation) throw new Error("Can't find sern.config.json");
+
+	const output = JSON.parse(await readFile(sernLocation, 'utf8')) as sernConfig;
+
+	if (!output) throw new Error("Can't read your sern.config.json.");
```
Comment (jacoobes) [inline src/utilities/getConfig.ts:11]: ive had the habit of using assert more for checking undefined.

its declarative and reads better

```ts
import assert from 'node:assert'

assert(output, "Can't read your sern.config.json.")
```

File: src/utilities/getConfig.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { readFile } from 'node:fs/promises';
+import { findUp } from 'find-up';
+
+export async function getConfig(): Promise<sernConfig> {
+	const sernLocation = await findUp('sern.config.json');
+
+	if (!sernLocation) throw new Error("Can't find sern.config.json");
```
Comment (jacoobes) [inline src/utilities/getConfig.ts:7]: assert(sernLocation, "Can't findyour sern.config.json.")

**Review by jacoobes — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (jacoobes) [reply src/index.ts:45]: ie: `sern publish src/**/*` works or `sern publish ping` works as well

Comment (EvolutionX-10) [reply src/index.ts:45]: will keep in mind

**Comment by jacoobes** [comment]
#106

**Review by jacoobes — DISMISSED** [review]

**Comment by EvolutionX-10** [comment]
fixed one bug and some minor issues
improved the spacing
and improved the ui for user (by adding messages and colors, and spinners)

**Review by jacoobes — APPROVED** [review]

**Pull request merged**